caffe2/python/core.py

```diff
@@ -2280,12 +2280,12 @@ def remap_input(op, blob_name_remapping):
 
 def copy_func_between_devices(src, dst):
     CPU = caffe2_pb2.CPU
-    CUDA = caffe2_pb2.CUDA
+    GPU = workspace.GpuDeviceType
 
     if src.device_type == CPU and dst.device_type == CPU:
         return None
 
-    if src.device_type == CUDA and dst.device_type == CUDA:
```

Comment:
I think here we should not compare to GpuDeviceType, because the logic of adding a Copy operator here is applicable as long as the src and dst are of the same gpu device type. I suggest let's add an util function IsGPUDeviceType:
```
def IsGPUDeviceType(device_type):
  return device_type in {caffe2_pb2.CUDA, caffe2_pb2.HIP}
```

and then change here to:
```
is_src_gpu = IsGPUDeviceType(src.device_type)
is_dst_gpu = IsGPUDeviceType(dst.device_type)
if is_src_gpu and is_dst_gpu:
  ...
```

Reply:
Got it!